src/download.php

```diff
@@ -87,10 +137,20 @@
         // Gene database contents.
         $sFileName = 'full_download_' . $_PE[3];
         $sHeader = 'Full data';
-        $sFilter = 'gene';
         $ID = $_PE[3];
         lovd_isAuthorized('gene', $_PE[3]);
-        lovd_requireAuth(LEVEL_CURATOR);
+
+        if ($_AUTH['level'] >= LEVEL_CURATOR) {
+            $sFilter = 'gene';
+        } else {
+            $bAllowDownload = $_DB->query('SELECT allow_download FROM ' . TABLE_GENES . ' WHERE id = ?', array($ID))->fetchColumn();
+            if ($bAllowDownload) {
+                $sFilter = 'gene_public';
+            } else {
+                die('Error: Data for gene (' . htmlspecialchars($ID) . ') is not public and you don\'t have permission '.
```

Comment:
I've added the `htmlspecialchars()` as a precaution, in case we ever drop the regexp on the ID. Beware of XSS attacks.